neurodiffeq/generators.py

```diff
@@ -255,9 +278,29 @@ def __init__(self, grid=(10, 10), xy_min=(0.0, 0.0), xy_max=(1.0, 1.0), method='
             grid_x, grid_y = torch.meshgrid(x, y, indexing='ij')
             self.grid_x, self.grid_y = grid_x.flatten(), grid_y.flatten()
             self.getter = lambda: (self.grid_x, self.grid_y)
+        elif method == 'chebyshev2-noisy':
+            self.getter = self.generate(xy_min, xy_max, grid, method='chebyshev2-noisy')
+        elif method == 'latin-hypercube':
```

Comment:
I am not familiar with latin hypercube sampling. It seems the x and y are equally-spaced-noisy composed with permutation. Then x and y are used to generate meshgrid. The permutation in this case doesn't do anything useful since the meshgrid acts as a cartesian product and includes every possible combination of (x_i, y_j).  

My understanding (it might be wrong) is that there is a restraint between x and y. They are not independent in the same that there are a sudoku-like restriction put in place. Can you confirm or correct me?

Reply:
This is exactly what the sampling is trying to do. So yeah, your interpretation is correct.